gufe/protocols/errors.py

```diff
@@ -0,0 +1,32 @@
+# This code is part of OpenFE and is licensed under the MIT license.
+# For details, see https://github.com/OpenFreeEnergy/gufe
+
+
+class BaseGufeError(Exception):
+    """The base gufe error that other errors should subclass."""
+
+
+# Protocol Errors
+class ProtocolValidationError(BaseGufeError):
+    """Error when the protocol setup or settings can not be validated."""
+
+
+class ProtocolSetupError(BaseGufeError):
+    """Error when executing the setup unit of the protocol."""
+
+
+class ProtocolExecutionError(BaseGufeError):
+    """Error when executing the production unit of the protocol."""
+
+
+class ProtocolAnalysisError(BaseGufeError):
+    """Error when trying to perform some analyses after the protocol has been executed."""
+
+
```

Comment:
May make sense to make a `ProtocolDAGResultError(ProtocolError)` class, and make the exceptions below subclass it?